chromium_src/chrome/browser/component_updater/widevine_cdm_component_installer.cc

```diff
@@ -3,39 +3,329 @@
+  } else {
+    bool success = ExtractArm64Dll(zip_path) && AddArm64ArchToManifest();
+    if (!success) {
+      result_ =
+          CrxInstaller::Result(update_client::InstallError::GENERIC_ERROR);
+    }
+  }
+  // The loader needs to be destroyed on the current sequence. We therefore
+  // need to do it here, and cannot wait until the destructor does it.
+  loader_.reset();
+  installed_.Signal();
+}
+
+bool Arm64DllInstaller::ExtractArm64Dll(base::FilePath zip_path) {
+  VLOG(2) << "Extracting Arm64 DLL.";
+  base::FilePath arm64_directory = GetPlatformDirectory(install_dir_);
+  base::File::Error error;
+  if (base::CreateDirectoryAndGetError(arm64_directory, &error)) {
+    if (!zip::Unzip(zip_path, arm64_directory)) {
+      LOG(ERROR) << "Failed to unzip Arm64 DLL.";
+      return false;
+    }
+  } else {
+    LOG(ERROR) << "Failed to create " << arm64_directory << ": "
+               << base::File::ErrorToString(error);
+    return false;
+  }
+  return true;
+}
+
+// Components contain a "manifest.json" file that has a list "accept_arch",
+// which lists the supported architectures. The file is checked on browser
+// startup when the components are registered. If the browser's architecture is
+// not in the list of supported architectures, then the component is
+// uninstalled. To prevent this from happening to our WIDEVINE_ARM64_DLL_FIX,
+// we need to add "arm64" to the list of supported architectures.
+bool Arm64DllInstaller::AddArm64ArchToManifest() {
+  VLOG(2) << "Adding Arm64 to manifest.json.";
+  base::FilePath manifest_path = install_dir_.AppendASCII("manifest.json");
+
+  std::string json_content;
+  if (!base::ReadFileToString(manifest_path, &json_content)) {
+    LOG(ERROR) << "Failed to read file: " << manifest_path;
+    return false;
+  }
+
+  absl::optional<base::Value> root = base::JSONReader::Read(json_content);
+  if (!root) {
+    LOG(ERROR) << "Failed to parse JSON.";
+    return false;
+  }
+
+  base::Value::Dict* root_dict = root->GetIfDict();
+  if (!root_dict) {
+    LOG(ERROR) << "Manifest is not a dictionary.";
+    return false;
+  }
+
+  base::Value* accept_arch = root_dict->FindByDottedPath("accept_arch");
+  if (!accept_arch) {
+    LOG(ERROR) << "Could not find accept_arch field.";
+    return false;
+  }
+
+  base::Value::List* accept_arch_list = accept_arch->GetIfList();
+  if (!accept_arch_list) {
+    LOG(ERROR) << "accept_arch is not a list.";
+    return false;
+  }
+
+  accept_arch_list->Append("arm64");
+
+  std::string new_json_content;
+  base::JSONWriter::Write(*root, &new_json_content);
+
+  if (!base::WriteFile(manifest_path, new_json_content)) {
+    LOG(ERROR) << "Failed to write file: " << manifest_path;
+    return false;
+  }
+
+  return true;
+}
+
+void RegisterWidevineCdmComponent(
+    ComponentUpdateService* cus,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+    base::OnceCallback<void()> callback) {
```

Comment:
nit:
```suggestion
    base::OnceClosure callback) {
```

Reply:
Resolved in 13eb3162c624bf579b296282181342dc23b79f60.